src/coreclr/src/gc/gc.cpp

```diff
@@ -17116,19 +17065,19 @@ void gc_heap::gc1()
 #endif // BIT64
                     gc_data_global.final_youngest_desired = desired_per_heap;
                 }
-#if 1 //subsumed by the linear allocation model
-                if (gen == (max_generation + 1))
+#if 1 //subsumed by the linear allocation model 
+                if (gen > max_generation)
```

Comment:
because you've changed everything to uoh below it would make sense to do 

`if (gen >= uoh_start_generation)` 

here